newIDE/app/src/ResourcesList/index.js

```diff
@@ -173,7 +172,13 @@ export default class ResourcesList extends React.Component<Props, State> {
       {
         label: 'Scan for Images',
         click: () => {
-          this._scanForNewResources();
+          this._scanForNewResources(IMAGE_EXTENSIONS, new gd.ImageResource());
```

Comment:
You should not pass a resource instance (`new gd.ImageResource()`) but the class Constructor (`gd.ImageResource`). Same for audio (`gd.AudioResource`).

Reply:
In fact, it might be better to pass a "creator function" (which would be useful if the creation of some resource is more complex or should be personalized):

`this._scanForNewResources(IMAGE_EXTENSIONS, () => new gd.ImageResource());`

The difference is small but important. Instead of passing the constructor (`gd.ImageResource`), you're passing a **function** that creates a gd.Resource (meaning that _scanForNewResources is not even aware of what is being created).

The advantage of this pattern is that if at some point we have complicated resource or if we want to add some default values to our resource, we can do it in the creator function:

```
this._scanForNewResources(IMAGE_EXTENSIONS,() => {
  const resource = new gd.ComplexResource();
  resource.set3DModel(something);
  return resource;
});
```

Reply:
that is very clever 👍  I will update it now :D